nmxact/xact/image.go

```diff
@@ -133,8 +133,14 @@ func findChunkLen(s sesn.Sesn, hash []byte, upgrade bool, data []byte,
 	off int, imageNum int, seq uint8) (int, error) {
 
 	// Let's start by encoding max allowed chunk len and we will see how many
-	// bytes we need to cut
-	chunklen := min(len(data)-off, IMAGE_UPLOAD_MAX_CHUNK)
+	// bytes we need to cut.
+	// First chunk needs to be header sized.
+	var chunklen int
+	if off == 0 {
```

Comment:
If we want to be super clean, we could add:
```
const IMAGE_UPLOAD_MAX_1ST_CHUNK = 512
```
alongside the already-existing `IMAGE_UPLOAD_MIN_1ST_CHUNK` at the top of the file and use that.

Reply:
Done